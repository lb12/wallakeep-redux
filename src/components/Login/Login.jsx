import React from "react";

import Tags from "../Tags/Tags";
import './Login.css';

export default class Login extends React.Component {
  constructor(props) {
    super(props);
    
    this.state = {
      user: { 
        firstname: '', 
        surname: '', 
        tag: '' 
      }
    };
  }
  
  onSubmit = evt => {
    evt && evt.preventDefault();
    this.props.setUser(this.state.user); // Save user into redux-store
    this.props.history.push('/'); //  Redirect user to home page always
  };

  onInputChange = evt => {
    const { name, value } = evt.target;

    this.setState(({ user }) => ({
      user: {
        ...user,
        [name]: value
      }
    }));
  };

  onTagSelected = optionSelected => {
    let user = this.state.user;
    user.tag = optionSelected;

    this.setState({ user });
  };

  render() {
    const { firstname, surname } = this.state.user;
    return (
      <div className="container login">
        <h1>Login</h1>
        <form className="mt-4" onSubmit={this.onSubmit}>
          <div className="form-group">
            <label htmlFor="firstname">Firstname</label>
            <input
              type="text"
              required
              name="firstname"
              value={firstname}
              className="form-control"
              placeholder="Enter firstname"
              onChange={this.onInputChange}
              id=""
            />
          </div>
          <div className="form-group">
            <label htmlFor="surname">Surname</label>
            <input
              type="text"
              required
              name="surname"
              value={surname}
              className="form-control"
              placeholder="Enter surname"
              onChange={this.onInputChange}
              id=""
            />
          </div>
          <div className="form-group">
            <label htmlFor="tags-select">Tags</label>
            <Tags required={true} onTagSelected={this.onTagSelected} />
          </div>
          <button type="submit" className="btn btn-primary mt-1">Login</button>
        </form>
      </div>
    );
  }
}