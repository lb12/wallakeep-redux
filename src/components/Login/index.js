import { connect } from 'react-redux';

import { setUser } from '../../store/actions';
import Login from './Login';
import withForm from '../Form/withForm';

const mapDispatchToProps = dispatch => ({
    setUser: user => dispatch( setUser(user) )
});

export default connect(null, mapDispatchToProps)(withForm(Login));