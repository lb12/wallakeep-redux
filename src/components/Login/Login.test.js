import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { createMemoryHistory } from 'history'
import Login from './Login';

// Enzyme configure
configure({ adapter: new Adapter() });

const defaultProps = {
    history: createMemoryHistory('/'), // necessary to simulate the redirection to '/'
    setUser: jest.fn()
};

describe('Login component test', () => {

    describe('Login render', () => {
        const render = props => shallow(<Login {...defaultProps} {...props} />);
        let wrapper;

        beforeEach(() => {
            wrapper = render();
        });


        it('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('should render', () => {
            expect(wrapper.exists()).toBe(true);
        });
       
        it('should call setUser on the "onSubmit" of the form', () => {
            const instance = wrapper.instance();
            wrapper.setState({tag: 'mobile'});
            instance.onSubmit({firstname: 'David', surname: 'Escribano'});      
            expect(defaultProps.setUser).toHaveBeenCalled();
        });
    });
}); 