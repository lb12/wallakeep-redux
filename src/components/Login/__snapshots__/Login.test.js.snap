// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component test Login render should render correctly 1`] = `
<div
  className="container login"
>
  <h1>
    Login
  </h1>
  <Form
    className="mt-4"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="firstname"
      >
        Firstname
      </label>
      <Input
        className="form-control"
        name="firstname"
        placeholder="Enter firstname"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="surname"
      >
        Surname
      </label>
      <Input
        className="form-control"
        name="surname"
        placeholder="Enter surname"
        required="required"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="tags-select"
      >
        Tags
      </label>
      <Tags
        onTagSelected={[Function]}
        required={true}
      />
    </div>
    <button
      className="btn btn-primary mt-1"
      type="submit"
    >
      Login
    </button>
  </Form>
</div>
`;
